### PR TITLE
FSF-3628 Planning fixes after subsystem meeting

### PR DESCRIPTION
**Pull request by lourenco31**

### CODE REVIEW

File: src/common_lib/include/common_lib/communication/marker.hpp [hunk 1]
```diff
@@ -107,4 +108,69 @@ visualization_msgs::msg::MarkerArray marker_array_from_structure_array(
...
+
+  for (size_t i = 0; i < structure_array.size(); ++i) {
+    geometry_msgs::msg::Point point;
+    point.x = structure_array[i].position.x;
+    point.y = structure_array[i].position.y,
+
+    marker.points.push_back(point);
+  }
+
+  marker.lifetime = rclcpp::Duration(std::chrono::duration<double>(5));
```
Comment (marhcouto) [inline src/common_lib/include/common_lib/communication/marker.hpp:171]: Remove this

File: src/planning/src/adapter_planning/eufs.cpp [hunk 2]
```diff
@@ -80,9 +80,8 @@ visualization_msgs::msg::MarkerArray marker_array_from_cone_array_w_covariance(
 }
 
 EufsAdapter::EufsAdapter(Planning* planning) : Adapter(planning) {
-  RCLCPP_INFO(this->node->get_logger(), "EufsAdapter created");
   if (this->node->using_simulated_se_) {
-    RCLCPP_INFO(this->node->get_logger(), "Eufs using simulated State Estimation\n");
+    RCLCPP_INFO(this->node->get_logger(), "Planning : EUFS using simulated State Estimation");
```
Comment (marhcouto) [inline src/planning/src/adapter_planning/eufs.cpp:84]: O prório logger já diz o nó que está a dar log

File: src/planning/src/planning/cone_coloring.cpp [hunk 3]
```diff
@@ -13,7 +13,11 @@ ConeColoring::ConeColoring(double gain_angle, double gain_distance, double gain_
       ncones_gain(gain_ncones),
       exponent1(exponent_1),
       exponent2(exponent_2),
-      max_cost(cost_max) {}
+      max_cost(cost_max) {
```
Comment (marhcouto) [inline src/planning/src/planning/cone_coloring.cpp:16]: Same here

File: src/planning/src/planning/planning.cpp [hunk 4]
```diff
@@ -43,32 +47,47 @@ Planning::Planning() : Node("planning") {
...
+    // Publisher for visualization
+    this->after_rem_yellow_cones_pub_ =
+        this->create_publisher<visualization_msgs::msg::MarkerArray>(
+            "/path_planning/after_rem_yellow_cones", 10);
   }
   // Publishes path from file in Skidpad & Acceleration events
   this->timer_ = this->create_wall_timer(
       std::chrono::milliseconds(100), std::bind(&Planning::publish_predicitive_track_points, this));
-  RCLCPP_INFO(this->get_logger(), "Using %s adapter", mode.c_str());
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Planning using mode: %s", mode.c_str());
```
Comment (marhcouto) [inline src/planning/src/planning/planning.cpp:62]: Why?

File: src/planning/src/planning/planning.cpp [hunk 5]
```diff
@@ -100,17 +119,25 @@ void Planning::track_map_callback(const custom_interfaces::msg::ConeArray &msg)
   auto path = triangulations_path;
   path_smoother->defaultSmoother(path);
 
-  RCLCPP_DEBUG(this->get_logger(), "Planning published %d path points", (int)path.size());
+  RCLCPP_DEBUG(rclcpp::get_logger("rclcpp"), "Planning published %i path points\n",
+               (int)path.size());
   publish_track_points(path);
```
Comment (marhcouto) [inline src/planning/src/planning/planning.cpp:124]: I would've separated this into another function, but no big deal

File: src/planning/src/planning/planning.cpp [hunk 6]
```diff
@@ -100,17 +119,25 @@ void Planning::track_map_callback(const custom_interfaces::msg::ConeArray &msg)
...
+                               cone_coloring->current_right_cones, triangulations_path, path,
+                               track->get_left_cones(), track->get_right_cones());
   }
 }
 
 void Planning::vehicle_localization_callback(const custom_interfaces::msg::VehicleState &msg) {
   this->pose = Pose(msg.position.x, msg.position.y, msg.theta);
+  if (this->recieved_first_track && !this->recieved_first_pose) {
+    this->recieved_first_pose = true;
+    run_planning_algorithms();
```
Comment (marhcouto) [inline src/planning/src/planning/planning.cpp:137]: Probably no need or gain in doing this on both callbacks, as these two are almost always published in pairs. Maybe using a message filter would be good in this case

File: src/planning/include/planning/planning.hpp [hunk 7]
```diff
@@ -63,21 +63,28 @@ class Planning : public rclcpp::Node {
...
+  std::vector<Cone *> cone_array;
   /**< Subscription to vehicle localization */
   rclcpp::Subscription<custom_interfaces::msg::VehicleState>::SharedPtr vl_sub_;
   /**< Subscription to track map */
   rclcpp::Subscription<custom_interfaces::msg::ConeArray>::SharedPtr track_sub_;
   /**< Local path points publisher */
   rclcpp::Publisher<custom_interfaces::msg::PathPointArray>::SharedPtr local_pub_;
   /**< Publisher for the final path*/
-  rclcpp::Publisher<visualization_msgs::msg::MarkerArray>::SharedPtr visualization_pub_;
+  rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr visualization_pub_;
```
Comment (marhcouto) [inline src/planning/include/planning/planning.hpp:76]: Why?

**Review by marhcouto — CHANGES_REQUESTED** [review]
Some changes requested

File: src/planning/include/planning/planning.hpp (hunk 7 above)
Comment (andrelds11) [reply src/planning/include/planning/planning.hpp:76]: It's now a Marker with a line, I'd say it's better than the list of markers, it differentiates more to the cones

**Review by marhcouto — APPROVED** [review]

**Pull request merged**